### PR TITLE
Wire agent config

### PR DESCRIPTION
**Pull request by jalvz**
## Motivation/summary

Read configuration from Elastic Agent and use the namespace for the indices

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

I have considered changes for:
- [ ] documentation
- [ ] logging (add log lines, choose appropriate log selector, etc.)
- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))
- [ ] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
- [ ] telemetry
- [ ] Elasticsearch Service (https://cloud.elastic.co)
- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)
- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)

### CODE REVIEW

File: publish/pub.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ var (
 // newPublisher creates a new publisher instance.
 //MaxCPU new go-routines are started for forwarding events to libbeat.
 //Stop must be called to close the beat.Client and free resources.
-func NewPublisher(pipeline beat.Pipeline, tracer *apm.Tracer, cfg *PublisherConfig) (*Publisher, error) {
+func NewPublisher(pipeline beat.Pipeline, namespace string, tracer *apm.Tracer, cfg *PublisherConfig) (*Publisher, error) {
```
Comment (axw) [inline publish/pub.go:84]: Add namespace as a field of PublisherConfig?

**Review by axw — COMMENTED** [review]
Looks good functionally, just a few minor things I'd like to see tidied up

File: beater/beater.go [hunk 2]
```diff
@@ -120,14 +121,27 @@ func (bt *beater) Run(b *beat.Beat) error {
...
 			var cfg *config.Config
-			cfg, err = config.NewConfig(ucfg.Config, elasticsearchOutputConfig(b))
+			apmServerCommonConfig := integrationConfig.APMServer
+			cfg, err = config.NewConfig(apmServerCommonConfig, elasticsearchOutputConfig(b))
 			if err != nil {
-				bt.logger.Warn("Could not parse configuration from Elastic Agent ", err)
+				bt.logger.Error("Could not parse apm-server configuration from Elastic Agent ", err)
+				return
 			}
+
```
Comment (axw) [inline beater/beater.go:139]: If you add the following (untested), you can remove the TODO above. This addresses the issue I mentioned on Slack, where the data streams config will be defined for the initial static config, but not for the dynamic input coming from Agent:

```go
apmServerCommonConfig.Merge(common.MustNewConfigFrom(`{"apm-server.data_streams.enabled": true}`))
```

**Review by axw — APPROVED** [review]

File: beater/beater.go (hunk 2 above)
Comment (jalvz) [reply beater/beater.go:139]: Not sure I follow, when agent starts the server it does it with `-E apm-server.data_streams.enabled=true`, right?

**Comment by axw** [comment]
Following on from https://github.com/elastic/apm-server/pull/4228#issuecomment-749328218, I sent a transaction to APM Server and checked that a new "traces" data stream was created with the namespace "default":

```
GET /_data_stream/traces-*
{
  "data_streams" : [
    {
      "name" : "traces-apm-default",
      "timestamp_field" : {
        "name" : "@timestamp"
      },
      "indices" : [
        {
          "index_name" : ".ds-traces-apm-default-2020.12.22-000001",
          "index_uuid" : "3AuDXYecSVOUbw_JwGDdQw"
        }
      ],
      "generation" : 1,
      "_meta" : {
        "managed_by" : "ingest-manager",
        "managed" : true,
        "package" : {
          "name" : "apm"
        }
      },
      "status" : "YELLOW",
      "template" : "traces-apm",
      "ilm_policy" : "traces",
      "hidden" : false
    }
  ]
}
```

I then edited the integration to set the namespace to "nondefault", restarted elastic-agent, and send another transaction. A new data stream was created:

```
GET /_data_stream/traces-*
{
  "data_streams" : [
    {
      "name" : "traces-apm-default",
      "timestamp_field" : {
        "name" : "@timestamp"
      },
      "indices" : [
        {
          "index_name" : ".ds-traces-apm-default-2020.12.22-000001",
          "index_uuid" : "3AuDXYecSVOUbw_JwGDdQw"
        }
      ],
      "generation" : 1,
      "_meta" : {
        "managed_by" : "ingest-manager",
        "managed" : true,
        "package" : {
          "name" : "apm"
        }
      },
      "status" : "YELLOW",
      "template" : "traces-apm",
      "ilm_policy" : "traces",
      "hidden" : false
    },
    {
      "name" : "traces-apm-nondefault",
      "timestamp_field" : {
        "name" : "@timestamp"
      },
      "indices" : [
        {
          "index_name" : ".ds-traces-apm-nondefault-2020.12.22-000001",
          "index_uuid" : "G9c-yRmBT3KR9-MLsu9vuQ"
        }
      ],
      "generation" : 1,
      "_meta" : {
        "managed_by" : "ingest-manager",
        "managed" : true,
        "package" : {
          "name" : "apm"
        }
      },
      "status" : "YELLOW",
      "template" : "traces-apm",
      "ilm_policy" : "traces",
      "hidden" : false
    }
  ]
}
```

**Pull request merged**